controllers/operatorconfig_controller.go

```diff
@@ -54,51 +76,343 @@ type OperatorConfigReconciler struct {
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
 func (r *OperatorConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := log.FromContext(ctx)
+	// Fetch the OperatorConfig operatorConfig
+	operatorConfig := &cranev1alpha1.OperatorConfig{}
+	if err := r.Get(ctx, req.NamespacedName, operatorConfig); err != nil {
+		// Error reading the object - requeue the request.
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			log.Info("OperatorConfig resource not found")
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Failed to get OperatorConfig")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
 
-	err := createClusterTasks(ctx, log)
+	if operatorConfig.Name != OwnerConfigName {
+		reason := fmt.Sprintf("Invalid name (%s)", operatorConfig.Name)
+		msg := fmt.Sprintf("Only one OperatorConfig supported per cluster and must be named '%s'", OwnerConfigName)
+		log.Info(fmt.Sprintf("%s: %s", reason, msg))
+		meta.SetStatusCondition(&operatorConfig.Status.Conditions, metav1.Condition{
+			Type:               InvalidConfigName,
+			Status:             True,
+			Reason:             "OperatorConfigNameInvalid",
```

Comment:
nit: change the to "NonStandardNameConfigured" ? to indicate why the InvalidName type condition was triggered to true.

More the guidelines of reason field can be found here https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions specifically 

> In condition types, and everywhere else they appear in the API, Reason is intended to be a one-word, CamelCase representation of the category of cause of the current statusIn condition types, and everywhere else they appear in the API, Reason is intended to be a one-word, CamelCase representation of the category of cause of the current status